src/client/components/AddMeal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from "react";
+import Header from "./Header";
+import Footer from "./Footer";
+
+export default function AddMeal() {
+  const [inputState, setInputstate] = useState(false);
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [location, setLocation] = useState("");
+  const [When, setWhen] = useState("");
+  const [reservations, setReservations] = useState("");
+  const [price, setPrice] = useState("");
+
+  function newMeal() {
+    {
+      if (title !== "") {
+        setInputstate(true);
+        fetch("/api/meals", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            title: title,
+            description: description,
+            location: location,
+            When_date: When,
+            max_reservations: reservations,
+            price: price,
+            created_date: new Date(),
+          }),
+        })
+          .catch((e) => {
+            setError(e);
+            alert("please enter the Valid details");
```

Comment:
I'd expect setting the error would render an error prompt in the page, so no need to alert. Browser alerts are very disrupting so I would avoid them! :)